### ISSUE TITLE
[Feature] Target interception

### ISSUE THREAD

**Issue by szmarczak** [microsoft/playwright#13112]
Previous issue: https://github.com/microsoft/playwright/issues/1101
Related: https://github.com/microsoft/playwright/issues/13111

There is `context.addInitScript`, however it doesn't solve all cases where target interception is needed.
For example, if there are CDP messages to be sent. I'm using CDP for [response interception](https://chromedevtools.github.io/devtools-protocol/tot/Fetch/#method-continueResponse), which cannot be done via Playwright yet.

In my opinion, there should be a way to intercept target creation, so Node.js code can be executed (and that enables a wide array of other possibilities) before webpage scripts are ran.

**Comment by rwoll** [comment]
> I'm using CDP for [response interception](https://chromedevtools.github.io/devtools-protocol/tot/Fetch/#method-continueResponse), which cannot be done via Playwright yet.

If your use case is response interception/modification, does https://playwright.dev/docs/network#modify-responses work for you?

**Comment by szmarczak** [comment]
Unfortunately it doesn't, as it uses Node.js HTTP API under the hood - this disables HTTP/2 - the behavior isn't the same. 

`Fetch.continueResponse` works for me. Ideally I'd like to capture all new targets, send some CDP messages, then resume target's work.

**Comment by yury-s** [comment]
> Unfortunately it doesn't, as it uses Node.js HTTP API under the hood - this disables HTTP/2 - the behavior isn't the same.
> 

Can you tell us more about your use case and where it fails with the current implementation but works with `Fetch.continueResponse`? 

> Fetch.continueResponse works for me.

We added that method during the first iteration on this feature but then moved away from it due to some technical issues. What we could do currently is to (optionally) use http2 instead of https but I'm not sure if that would address your problem given the requirements you mentioned above, we need to better understand your scenario.

**Comment by szmarczak** [comment]
FYI target interception is currently blocked by https://crbug.com/1070568

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
